clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1443,9 +1449,12 @@ class KernelObjVisitor {
                   HandlerTys &...Handlers) {
     if (isSyclSpecialType(ParamTy, SemaSYCLRef))
       KP_FOR_EACH(handleOtherType, Param, ParamTy);
-    else if (ParamTy->isStructureOrClassType())
-      KP_FOR_EACH(handleOtherType, Param, ParamTy);
-    else if (ParamTy->isUnionType())
+    else if (ParamTy->isStructureOrClassType()) {
+      if (KF_FOR_EACH(handleStructType, Param, ParamTy)) {
+        CXXRecordDecl *RD = ParamTy->getAsCXXRecordDecl();
+        visitRecord(RD, Param, RD, ParamTy, Handlers...);
```

Comment:
Yeah, a lot of code around the visitors is intended to have some "owner" because the original use case is lambda/functor whose fields need to be visited. However it doesn't seem to affect things anyhow and I suspect not having it is fine. Also this comment
https://github.com/intel/llvm/blob/729d6f62ff2e0676f01d58d256392bd94e7b71f7/clang/lib/Sema/SemaSYCL.cpp#L1301
suggests so.
I transformed this argument to `nullptr` to avoid confusion.